packages/integration-tests/migrations/1580658856631_author.ts

```diff
@@ -190,6 +181,18 @@ export function up(b: MigrationBuilder): void {
     publisher_id: foreignKey("publishers", { notNull: false }),
   });
 
+  createEntityTable(b, "users", {
+    name: { type: "varchar(255)", notNull: true },
+    email: { type: "varchar(255)", notNull: true },
+    // for testing o2o/m2o renames
+    author_id: foreignKey("authors", {
+      notNull: false,
+      unique: true,
+      oneToOneName: "userOneToOne",
+      referenceName: "authorManyToOne",
```

Comment:
Thought: I have the terms:

* `fieldName: "authorManyToOne"`, and
* `otherFieldName: "userOneToOne"`

Burned in my head as what I think of these; do you mind renaming?

My thought is that `fieldName` is just a generalization of `referenceName`, and other `otherFieldName` is similarly a generalization of "the name on the other side, regardless of whether the other side is `o2o`/`m2o`".

(Seems nice to have the user not really care whether they need to use `oneToOneName` or `oneToManyName` based on the `unique`-ness of the FK.)

Reply:
This way corresponds to what the codegen calls them though

Reply:
I think i'm about a 5/10 so i'll change it